src/pythoninterpretermodule.cpp

```diff
@@ -54,13 +53,16 @@ void PythonInterpreterModule::shutdown() {
 
     state = ModuleState::STARTED_SHUTDOWN;
     SPDLOG_INFO("{} shutting down", PYTHON_INTERPRETER_MODULE_NAME);
-    py::gil_scoped_acquire acquire;
+    this->GILExpulsion.reset();
     if (pythonBackend != nullptr)
         delete pythonBackend;
     state = ModuleState::SHUTDOWN;
     SPDLOG_INFO("{} shutdown", PYTHON_INTERPRETER_MODULE_NAME);
     py::finalize_interpreter();
 }
+void PythonInterpreterModule::exileGILFromCurrentThread() const {
+    this->GILExpulsion = std::make_unique<py::gil_scoped_release>();
+}
```

Comment:
Yeah, I'm thinking about the naming.... as for the method name I think `releaseGIL()` would be enough.
The field on the other hand. Maybe we could call it `GILReleaseHandle` and create additional method that will reset that pointer like: `requireGIL()` and call it in line 56.

Reply:
Done